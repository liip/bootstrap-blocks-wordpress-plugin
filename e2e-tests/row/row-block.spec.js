/**
 * WordPress dependencies
 */
import {
	createNewPost,
	getEditedPostContent,
	clickBlockToolbarButton,
	clickButton,
} from '@wordpress/e2e-test-utils';
import { insertRowBlock, selectRowBlock } from './row-helper';
import { clickElementByText } from '../helper';

describe( 'row block', () => {
	beforeEach( async () => {
		await createNewPost();
	} );

	it( 'Row block should be available', async () => {
		await insertRowBlock();

		// Check if row block was inserted
		expect(
			await page.$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/row"]'
			)
		).not.toBeNull();
		expect(
			await page.$$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/column"]'
			)
		).toHaveLength( 2 );

		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	/* TODO fix broken test
	it( 'Should be possible to change alignment', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Change horizontal alignment
		await clickBlockToolbarButton(
			'Change horizontal alignment of columns'
		);
		await clickButton( 'Align columns right' );
		expect(
			await page.$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/row"][data-alignment="right"]'
			)
		).not.toBeNull();
		expect( await getEditedPostContent() ).toMatchSnapshot();

		// Change vertical alignment
		await clickBlockToolbarButton( 'Change vertical alignment of columns' );
		await clickButton( 'Align columns bottom' );
		expect(
			await page.$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/row"][data-vertical-alignment="bottom"]'
			)
		).not.toBeNull();
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
	 */

	it( 'Should be possible to change column layout', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Layout options should be visible
		expect(
			await page.$$( '.wp-bootstrap-blocks-template-selector-button' )
		).toHaveLength( 5 );
		expect(
			await page.$(
				'.wp-bootstrap-blocks-template-selector-button > button[aria-label="2 Columns (1:1)"].is-active'
			)
		).not.toBeNull();
		expect(
			await page.$$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/column"][data-size-md="6"]'
			)
		).toHaveLength( 2 );

		// Template should be applied
		await page.click(
			'.wp-bootstrap-blocks-template-selector-button > button[aria-label="3 Columns (1:1:1)"]'
		);
		await page.waitFor( 1000 );
		expect(
			await page.$$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/column"][data-size-md="4"]'
			)
		).toHaveLength( 3 );
		expect( await getEditedPostContent() ).toMatchSnapshot();
		await page.click(
			'.wp-bootstrap-blocks-template-selector-button > button[aria-label="2 Columns (2:1)"]'
		);
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Should be possible to select custom template', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Custom template should add block list appender (shouldn't change current layout)
		await page.click(
			'.wp-bootstrap-blocks-template-selector-button > button[aria-label="Custom"]'
		);
		expect(
			await page.$(
				'.wp-block-wp-bootstrap-blocks-row > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-list-appender'
			)
		).not.toBeNull();
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );

	it( 'Should only be possible to add column in custom layout', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Select custom template
		await page.click(
			'.wp-bootstrap-blocks-template-selector-button > button[aria-label="Custom"]'
		);
		expect(
			await page.$(
				'.wp-block-wp-bootstrap-blocks-row > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-list-appender'
			)
		).not.toBeNull();

		// Add column block by clicking the block list appender
		await page.click(
			'.wp-block-wp-bootstrap-blocks-row > .block-editor-inner-blocks > .block-editor-block-list__layout > .block-list-appender'
		);
		const numberOfColumnBlocks = (
			await page.$$(
				'.block-editor-block-list__block[data-type="wp-bootstrap-blocks/column"]'
			)
		 ).length;
		expect( numberOfColumnBlocks ).toEqual( 3 );
	} );

	it( 'Should be possible to apply row options', async () => {
		await insertRowBlock();
		await selectRowBlock();

		// Enable no gutters option
		await clickElementByText( 'label', 'No Gutters' );
		expect( await getEditedPostContent() ).toMatchSnapshot();
	} );
} );
