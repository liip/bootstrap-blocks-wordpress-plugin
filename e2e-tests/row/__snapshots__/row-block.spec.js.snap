// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`row block Row block should be available 1`] = `
"<!-- wp:wp-bootstrap-blocks/row -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block Should be possible to apply row options 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"noGutters\\":true} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block Should be possible to change column layout 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"template\\":\\"1-1-1\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block Should be possible to change column layout 2`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"template\\":\\"2-1\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":8} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":4} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;

exports[`row block Should be possible to select custom template 1`] = `
"<!-- wp:wp-bootstrap-blocks/row {\\"template\\":\\"custom\\"} -->
<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->

<!-- wp:wp-bootstrap-blocks/column {\\"sizeMd\\":6} /-->
<!-- /wp:wp-bootstrap-blocks/row -->"
`;
